### PR TITLE
Creates a fresh log sub-directory on each run

### PR DESCRIPTION
**Pull request by m-harrison**
#125 
--
This proposed change takes a dependency on `uuidm` to generate UUIDs. It extends the `alcotest` global state to hold a globally unique run ID. This ID is newly generated on each call into `run_with_args` and is displayed out to the user before running any tests. The output logs from the tests are then placed in this sub directory which itself is under the `-o` directory (or under the default test directory if this is not specified).

To test this change I pinned my local clone with the changes, built the test executable of another project and repeatedly manually invoked it. The output logs were subsequently generated in separate sub-directories corresponding to the ID of the run, as expected.

#135 
--
This was caused by `dune` altering the current working directory before running tests. It would not explicitly pass in a `-o` parameter and so the default one was taken which was a relative path. The result of this was that a relative path was printed to the user and while this was correct it wasn't obvious that it was from a different current working directory. My fix made the default parameter an absolute path under the current working directory with the same relative path (functionally this is a no-op change). This meant when printed to the user it is now correct.

### CODE REVIEW

File: src/alcotest.ml [hunk 1]
```diff
@@ -383,7 +390,9 @@ let show_result t result =
     in
     let full_logs ppf =
       if t.verbose then Fmt.string ppf ""
-      else Fmt.pf ppf "The full test results are available in `%s`.\n" t.test_dir
+      else
+        Fmt.pf ppf "The full test results are available in `%s`.\n"
+          (output_dir t)
```
Comment (samoht) [inline src/alcotest.ml]: Not directly related to that PR but #135 points that this location is in general wrong. Might be a good occasion to fix that too, but I am fine if this is done in a different PR too!

Comment (m-harrison) [reply src/alcotest.ml]: Good call out, see the comment I left on #135 for my analysis of the issue. The summary is that I believe `dune` alters the current working directory and as the default `test_dir` is a relative rather than an absolute path it appears to be wrong (the problem being that in the context of the test runner it is correct).

My fix (pushed in `fbfb395`) is to set this default `test_dir` to an absolute path whose prefix is the current working directory. This means semantically there is no change but when `t.test_dir` is printed to the user it is explicit about the current working directory (and just the entire location). It also means the current working directory is taken once at the start of the run rather than implicitly picking it up every time that `t.test_dir` was referenced in some I/O.

**Comment by m-harrison** [comment]
> That looks great, thanks a lot for your patches! I've just pushed a fix for the CI, hopefully everything should be green now.

Thanks for the review! It looks like the CI is still red so I'll leave that with you. I don't have write access to the repo so will need you to merge anyway. If you want to keep this one open a while to try and get the CI working then feel free!

**Comment by samoht** [comment]
Thanks a lot for your patch! I'll draft a new release with your changes.
